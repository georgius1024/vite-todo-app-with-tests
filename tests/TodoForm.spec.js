import { describe, it, expect } from "vitest";
import { mount } from "@vue/test-utils";
import component from "src/components/TodoForm.vue";

describe("TodoForm component", () => {
  it("mounts without error", () => {
    const wrapper = mount(component);
    expect(wrapper).toBeDefined();
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('handles input and emits events', () => {
    const wrapper = mount(component);
    const inputControl = wrapper.find('[data-testid="item-input"]')
    const buttonControl = wrapper.find('[data-testid="item-submit"]')
    inputControl.setValue('new item')
    buttonControl.trigger('click')
    expect(wrapper.emitted().add).toHaveLength(1)
    expect(wrapper.emitted().add).toHaveProperty("0.0", "new item")
  })

});
