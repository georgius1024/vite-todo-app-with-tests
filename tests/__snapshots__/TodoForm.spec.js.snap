// Vitest Snapshot v1

exports[`TodoForm component > mounts without error 1`] = `"<div data-v-2bcb7aa7=\\"\\" class=\\"form\\"><input data-v-2bcb7aa7=\\"\\" placeholder=\\"enter todo\\" data-testid=\\"item-input\\"><button data-v-2bcb7aa7=\\"\\" data-testid=\\"item-submit\\">Add</button></div>"`;
